### ISSUE TITLE
`migadu_mailbox` resource fails when there are any identies

### ISSUE THREAD

**Issue by ajgon** [metio/terraform-provider-migadu#145]
When trying to update a mailbox containing identities a following error is thrown:

```
Error: Error Reading Mailbox

  with module.mail.migadu_mailbox.ajgon,
  on mail/mailboxes.tf line 1, in resource "migadu_mailbox" "ajgon":
   1: resource "migadu_mailbox" "ajgon" {

While calling the API, an unexpected error was returned in the response. Please contact the provider developer if you are unsure how to resolve the error.

Error: GetMailbox: json: cannot unmarshal object into Go struct field Mailbox.identities of type string
```

I think the problem is, the plugin expects identities to be an array of strings, while API returns objects, i.e.

```json
{
  "mailboxes": [
    {
      ...
      "identities": [
        {
          "name": "Identity",
          "address": "identity@example.com",
          "domain_name": "example.com",
          "local_part": "identity",
          "may_receive": true,
          "may_send": true,
          "may_access_imap": true,
          "may_access_managesieve": true,
          "may_access_pop3": true
        }
      ],
      ...
    }
  ]
}
```

**Comment by sebhoss** [comment]
Thanks for opening this ticket & sorry that this is not working right now. Not sure right now why I declared them as strings and will check & change that once I'm back in front of a computer

**Comment by sebhoss** [comment]
Ok a fix for this was just released with version [2024.2.9](https://github.com/metio/terraform-provider-migadu/releases/tag/2024.2.9). I have opted to simply remove the identities from mailbox models since we do have separate resources to manage identities anyway..

That said, I think I have found some problems with the identity API provided by Migadu, e.g. creating an identity with a password through the API does not seem to work as expected. I have contacted their support, but I would recommend to create identities using their web UI for now and import them later into terraform.

**Comment by ajgon** [comment]
Awesome, thank you!

**Comment by ajgon** [comment]
Hmm... I'm stilling having the same issue, after upgrade:

```
Error: GetMailbox: json: cannot unmarshal object into Go struct field Mailbox.identities of type string
```

I also double-checked that opentofu did proper upgrade:

```bash
$ ls .terraform/providers/registry.opentofu.org/metio/migadu/
2024.2.9
```

Maybe it's a problem with underlying migadu-client library? I'm happy to provide more debug info if you tell me how :)

**Comment by sebhoss** [comment]
oops yeah you are right, I was trying multiple ways to work on this and ended up not pulling in the latest version of the client. I'll fix that and release a new version tomorrow. Sorry for the inconvenience!

**Comment by sebhoss** [comment]
Ok so lets try this again - release [2024.2.10](https://github.com/metio/terraform-provider-migadu/releases/tag/2024.2.10) contains an updated client. I have tested this locally with a mailbox that has multiple identities and I could import/modify it just fine.

Additionally, I heard back from Migadu support and they told me that they are going to update their documentation wrt. identity creation. The missing piece here was the undocumented `password_use` parameter which controls how the newly created identity is accessed. I've added some examples in https://registry.terraform.io/providers/metio/migadu/latest/docs/resources/identity in case you are interested in managing identities.

Thanks for taking your taking your time in opening this ticket. It improved not only this provider, but the Migadu docs as well :heart:

**Comment by ajgon** [comment]
Anytime, I'm happy to help :)

Did another set of checks - now everything works like a charm, thank you! 

One last question though - does it make sense to make `password` (in identities) field required when `password_use` is `none`? Or is it a limitation of migadu API?

**Comment by sebhoss** [comment]
Argh yeah you are right again. I did not realize that `password` is currently required even though it is only really required if `password_use` is set to `custom`. In all other cases, the password can be set, but the Migadu API will ignore it.

I'll open up another PR to fix that :sweat_smile:

**Comment by sebhoss** [comment]
The change wrt. password validation was released with [2024.2.11](https://github.com/metio/terraform-provider-migadu/releases/tag/2024.2.11)

**Comment by ajgon** [comment]
> The change wrt. password validation was released with [2024.2.11](https://github.com/metio/terraform-provider-migadu/releases/tag/2024.2.11)

And it works great, thank you! :)

**Comment by sebhoss** [comment]
Nice! Thanks for the feedback & all your time improving this provider ^^
